src/XMakeTasks/AssemblyDependency/AssemblyInformation.cs

```diff
@@ -317,8 +329,102 @@ private FrameworkName GetFrameworkName()
             }
 
             return frameworkAttribute;
+#else
+            using (var stream = File.OpenRead(_sourceFile))
+            using (var peFile = new PEReader(stream))
+            {
+                var metadataReader = peFile.GetMetadataReader();
+
+                var attrs = metadataReader.GetAssemblyDefinition().GetCustomAttributes()
+                    .Select(ah => metadataReader.GetCustomAttribute(ah));
+
+                foreach (var attr in attrs)
+                {
+                    var ctorHandle = attr.Constructor;
+                    if (ctorHandle.Kind != HandleKind.MemberReference)
+                    {
+                        continue;
+                    }
+
+                    var container = metadataReader.GetMemberReference((MemberReferenceHandle)ctorHandle).Parent;
+                    var name = metadataReader.GetTypeReference((TypeReferenceHandle)container).Name;
+                    if (!string.Equals(metadataReader.GetString(name), "TargetFrameworkAttribute"))
+                    {
+                        continue;
+                    }
+
+                    var arguments = GetFixedStringArguments(metadataReader, attr);
+                    if (arguments.Count == 1)
+                    {
+                        return new FrameworkName(arguments[0]);
+                    }
+
+                }
+            }
+            return null;
+#endif
         }
 
+#if !FEATURE_ASSEMBLY_LOADFROM
+        //  This method copied from DNX source: https://github.com/aspnet/dnx/blob/e0726f769aead073af2d8cd9db47b89e1745d574/src/Microsoft.Dnx.Tooling/Utils/LockFileUtils.cs#L385
+        /// <summary>
+        /// Gets the fixed (required) string arguments of a custom attribute.
+        /// Only attributes that have only fixed string arguments.
+        /// </summary>
+        private static List<string> GetFixedStringArguments(MetadataReader reader, CustomAttribute attribute)
+        {
+            // TODO: Nick Guerrera (Nick.Guerrera@microsoft.com) hacked this method for temporary use.
+            // There is a blob decoder feature in progress but it won't ship in time for our milestone.
+            // Replace this method with the blob decoder feature when later it is availale.
```

Comment:
I actually talked with Nick about this.  The functionality is available in an alpha version of System.Reflection.Metadata, but the APIs still need to be reviewed and will probably change, so I preferred to use code that wouldn't break when that happens.